### PR TITLE
Add JavaPoet interop docs

### PR DESCRIPTION
**Pull request by ZacSweers**

### CODE REVIEW

File: docs/interop-javapoet.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+KotlinPoet-javapoet
```
Comment (Egorand) [inline docs/interop-javapoet.md:1]: nit: same as with KSP, maybe `JavaPoet extensions for KotlinPoet`?

File: docs/interop-javapoet.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+===================
+
+`interop:ksp` is an interop API for converting[JavaPoet](https://github.com/squareup/javapoet)
```
Comment (Egorand) [inline docs/interop-javapoet.md:4]: nit:
```suggestion
`interop:ksp` is an interop API for converting [JavaPoet](https://github.com/squareup/javapoet)
```

File: docs/interop-javapoet.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+```kotlin
+// Points to com.squareup.kotlinpoet.TypeName
+KTypeName
+// Points to com.squareup.javapoet.TypeName
+JTypeName
+```
+
+**Convert between a `JTypeName` and `KTypeName`**
+
+Most usage of these can run through the `toKTypeName()` and `toJTypeName()` extensions.
```
Comment (Egorand) [inline docs/interop-javapoet.md:24]: nit:
```suggestion
Most usages of these can run through the `toKTypeName()` and `toJTypeName()` extensions.
```

**Review by Egorand — APPROVED** [review]

File: docs/interop-javapoet.md (hunk 1 above)
Comment (ZacSweers) [reply docs/interop-javapoet.md:1]:
```suggestion
JavaPoet Extensions for KotlinPoet
```

File: docs/interop-javapoet.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+JavaPoet Extensions for KotlinPoet
+===================
```
Comment (ZacSweers) [inline docs/interop-javapoet.md:2]:
```suggestion
JavaPoet Extensions for KotlinPoet
==================================
```

File: docs/interop-javapoet.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+`interop:ksp` is an interop API for converting [JavaPoet](https://github.com/squareup/javapoet)
```
Comment (ZacSweers) [inline docs/interop-javapoet.md:4]:
```suggestion
`interop:javapoet` is an interop API for converting [JavaPoet](https://github.com/squareup/javapoet)
```

**Pull request merged**